iroh/src/node/builder.rs

```diff
@@ -256,7 +260,11 @@ where
     }
 
     /// Configure rpc endpoint, changing the type of the builder to the new endpoint type.
-    pub fn rpc_endpoint<E2: ServiceEndpoint<RpcService>>(self, value: E2) -> Builder<D, E2> {
+    pub fn rpc_endpoint<E2: ServiceEndpoint<RpcService>>(
```

Comment:
Not for this PR probably, but I guess we don't need this whole type parameter anymore on the builder due to having the ability to box the endpoint.

So far we don't make use of this on the node side but just on the client side, but maybe something to do for the next release...

We need to get rid of type parameters to make room for new type parameters... 😁